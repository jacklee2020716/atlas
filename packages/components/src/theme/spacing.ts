export default {
  xxs: "2px",
  xs: "4px",
  s: "8px",
  m: "16px",
  l: "20px",
  xl: "24px",
  xxl: "32px",
  xxxl: "40px",
  "4l": "48px",
  "5l": "64px"
}
