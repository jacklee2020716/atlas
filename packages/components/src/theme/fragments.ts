import { css } from "@emotion/core";
import spacing from "./spacing";
import typography from "./typography";

export function withSize(size: string) {
  return css`
    padding: ${size === "large"
      ? `${spacing.s4} ${spacing.s2}`
      : size === "normal" || size === "full"
      ? `${spacing.s3} ${spacing.s2}`
      : `${spacing.s2} ${spacing.s1}`};

    font-size: ${size === "large"
      ? typography.sizes.large
      : size === "normal" || size === "full"
      ? typography.sizes.normal
      : typography.sizes.small};

    width: ${size === "full" ? "100%" : "auto"};
  `;
}
