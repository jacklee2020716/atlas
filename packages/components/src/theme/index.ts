import typography from "./typography"
import colors from "./colors"
import spacing from "./spacing"
import breakpoints from "./breakpoints"
import * as fragments from "./fragments"

export { typography, breakpoints, spacing, colors, fragments }
