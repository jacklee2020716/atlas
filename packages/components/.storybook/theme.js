import { create } from "@storybook/theming/create";

import { colors } from "../src/theme";

export default create({
	base: "dark",

	colorPrimary: "#4038FF",
	colorSecondary: "deepskyblue",

	// UI
	appBg: colors.black,
	appContentBg: "#272D33",
	inputBg: "black",
	appBorderColor: "#424E57",
	appBorderRadius: 4,

	// Typography
	fontBase: '"Open Sans", sans-serif',
	fontCode: "monospace",

	// Text colors
	textColor: "#DAE2EB",
	textInverseColor: "rgba(255,255,255,0.9)",

	// Toolbar default and active colors

	barBg: "#272D33",
	barTextColor: "#7B8A95",
	barSelectedColor: "white",

	// Form colors
	inputBg: "white",
	inputBorder: "#272D33",
	inputTextColor: "white",
	inputBorderRadius: 4,

	brandTitle: "@joystream/components",
	brandUrl: "https://joystream.org",
	brandImage:
		"https://raw.githubusercontent.com/Joystream/design/master/logo/logo/PNG/Logo-horisontal-basic-white-1x.png",
});
