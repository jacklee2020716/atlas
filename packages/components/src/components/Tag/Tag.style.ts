import { css } from "@emotion/core"
import { typography, colors } from "../../theme"

export type TagStyleProps = {}

export let makeStyles = ({}: TagStyleProps) => {
  return css`
    border: 1px solid ${colors.blue[700]};
    color: ${colors.white};
    background-color: ${colors.black};
    text-align: center;
    padding: 10px 15px;
    display: inline-block;
    cursor: default;
    font-family: ${typography.fonts.base};
    font-weight: ${typography.weights.regular};
    font-size: ${typography.sizes.button.medium};
    margin: 0 15px 0 0;

    &::selection {
      background: transparent;
    }
  `
}
