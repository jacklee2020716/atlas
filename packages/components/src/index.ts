import * as theme from "./theme"

export { default as Button } from "./components/Button"
export { default as Carousel } from "./components/Carousel"
export { default as Dropdown } from "./components/Dropdown"
export { default as Grid } from "./components/Grid"
export { default as Header } from "./components/Header"
export { default as Link } from "./components/Link"
export { default as NavButton } from "./components/NavButton"
export { default as RadioButton } from "./components/RadioButton"
export { default as Checkbox } from "./components/Checkbox"
export { default as TagButton } from "./components/TagButton"
export { default as Tabs } from "./components/Tabs"
export { default as Tab } from "./components/Tabs/Tab"
export { default as Tag } from "./components/Tag"
export { default as TextField } from "./components/TextField"
export { default as Typography } from "./components/Typography"
export { default as VideoPreview } from "./components/VideoPreview"
export { default as VideoPlayer } from "./components/VideoPlayer"
export { default as SeriesPreview } from "./components/SeriesPreview"
export { default as ChannelPreview } from "./components/ChannelPreview"
export { theme }
