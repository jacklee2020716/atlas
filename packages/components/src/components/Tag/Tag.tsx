import React from "react"
import { makeStyles, TagStyleProps } from "./Tag.style"

type TagProps = {
  text: string
} & TagStyleProps

export default function Tag({
  text,
  ...styleProps
}: TagProps) {
  let styles = makeStyles(styleProps)
  return (
    <div css={styles}>
      {text}
    </div>
  )
}
