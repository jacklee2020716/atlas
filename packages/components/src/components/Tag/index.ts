import { memo } from "react"
import Tag from "./Tag"
export default memo(Tag)
