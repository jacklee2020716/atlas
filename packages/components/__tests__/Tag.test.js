import React from "react"
import { mount } from "enzyme"
import Tag from "./../src/components/Tag"
import { faCheck } from "@fortawesome/free-solid-svg-icons"
import { colors } from "./../src/theme"

describe("Tag component", () => {

  const component = mount(<Tag icon={faCheck} text="test" color={colors.other.success} />)

  it("Should render correctly", () => {
    expect(component).toMatchSnapshot()
  })

  it("Should render icon.", () => {
    expect(component.find("svg"))
      .toHaveLength(1)
  })

  it("Should render text.", () => {
    expect(component.contains(<span>test</span>))
      .toBe(true)
  })

})