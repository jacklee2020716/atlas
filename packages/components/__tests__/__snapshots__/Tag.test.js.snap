// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag component Should render correctly 1`] = `
<Memo(Tag)
  color="#219653"
  icon={
    Object {
      "icon": Array [
        512,
        512,
        Array [],
        "f00c",
        "M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z",
      ],
      "iconName": "check",
      "prefix": "fas",
    }
  }
  text="test"
>
  <div
    css={
      Object {
        "map": undefined,
        "name": "1wt924n",
        "next": undefined,
        "styles": "
      display: inline-block;
      font-family: 'Lato','Helvetica Neue',Arial,Helvetica,sans-serif;
      border: 1px solid #219653;
      border-radius: 4px;
      padding: 5px 10px;
      color: #219653;
      cursor: default;
      vertical-align: middle;
    ",
        "toString": [Function],
      }
    }
  >
    <FontAwesomeIcon
      border={false}
      className=""
      css={
        Object {
          "map": undefined,
          "name": "1nzu8hs",
          "next": undefined,
          "styles": "
      & > path:nth-of-type(1) {
        color: inherit;
        flex-shrink: 0;
      }
    ",
          "toString": [Function],
        }
      }
      fixedWidth={false}
      flip={null}
      icon={
        Object {
          "icon": Array [
            512,
            512,
            Array [],
            "f00c",
            "M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z",
          ],
          "iconName": "check",
          "prefix": "fas",
        }
      }
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      swapOpacity={false}
      symbol={false}
      title=""
      transform={null}
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-check fa-w-16 "
        css={
          Object {
            "map": undefined,
            "name": "1nzu8hs",
            "next": undefined,
            "styles": "
      & > path:nth-of-type(1) {
        color: inherit;
        flex-shrink: 0;
      }
    ",
            "toString": [Function],
          }
        }
        data-icon="check"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </FontAwesomeIcon>
    <span
      style={
        Object {
          "margin": "0 10px 0 0",
        }
      }
    />
    <span>
      test
    </span>
  </div>
</Memo(Tag)>
`;
